src/cascadia/TerminalApp/AppKeyBindings.idl

```diff
@@ -50,6 +50,9 @@ namespace TerminalApp
     {
         AppKeyBindings();
 
+        Windows.Data.Json.JsonArray ToJson();
+        static AppKeyBindings FromJson(Windows.Data.Json.JsonArray json);

```

Comment:
do these have to be in the IDL? they can also just be in the private/internal impl class version, which other components can know about

Reply:
I'm not sure - it's possible, though I was worried that since we're stashing the variable as just a `winrt::TerminalApp::AppKeybindings`, that cppwinrt would forget the actual implementation type. We'd probably have to change most of the `winrt::TerminalApp::AppKeybindings`'s to `winrt::TerminalApp::implementation::AppKeybindings`

Reply:
that's fine. there's also a way to make cppwinrt pull out the impl class on your behalf so you can transit it as the runtimeclass interface and bounce back to impl when necessary

Reply:
[get_self](https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/get-self)

Reply:
I tried this out, but it ended up feeling less clean than leaving them on the idl. I think for winrt-y types, we should probably stick with defining things in the idl as much as possible. I'm not really sure how the resource management works for a winrt type's implementation. Is it still ref-counted? Or is it just a normal stack object?

Reply:
I think it's ref counted through its owning object. You `get_self` on it, which gives you a temporary ref to its internals, but the ownership still lies with the owning ref you called `get_self` from